titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/AmazonClientProvider.java

```diff
@@ -0,0 +1,65 @@
+package com.netflix.titus.ext.aws;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.amazonaws.arn.Arn;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingAsync;
+import com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingAsyncClientBuilder;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceAsync;
+
+@Singleton
+public class AmazonClientProvider {
+
+    private final AwsConfiguration configuration;
+    private final AWSSecurityTokenServiceAsync stsClient;
+
+    private ConcurrentMap<String, AWSCredentialsProvider> awsCredentialsByAccountId = new ConcurrentHashMap<>();
+    private ConcurrentMap<String, AmazonElasticLoadBalancingAsync> loadBalancerClients = new ConcurrentHashMap<>();
+
+    @Inject
+    public AmazonClientProvider(AwsConfiguration configuration,
+                                AWSSecurityTokenServiceAsync stsClient) {
+        this.configuration = configuration;
+        this.stsClient = stsClient;
+    }
+
+    public AmazonElasticLoadBalancingAsync getLoadBalancingClient(String accountId) {
+        return loadBalancerClients.computeIfAbsent(accountId, id -> {
```

Comment:
Is it ok to create many of those during initialization if there are concurrent calls?

Reply:
:+1: probably not a huge deal, it might just increase the number of thread (pools) and connections during initialization a bit, but I agree it would be probably better to do a simple double-checked lock of the whole thing here and serialize calls

Reply:
Fixed